@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -43,12 +44,12 @@ function checkViewDistance(
         return textDistance <= maxViewDistance ? textDistance : undefined;
     }
 
-    // Spherical projection
+    // For sphere projection: Filter labels that are close to the horizon
     tmpPosition.copy(textElement.position).normalize();
-    camera.getWorldDirection(tmpCameraDir);
+    camera.getWorldPosition(tmpCameraDir).normalize();
+    const alpha = Math.acos(tmpPosition.dot(tmpCameraDir));
 
-    // TODO: Revisit, why is this angle check needed and where does the constant -0.6 come from?
-    return tmpPosition.dot(tmpCameraDir) < -0.6 && textDistance <= maxViewDistance
+    return alpha < TEXT_ELEMENT_FALLOFF_ANGLE && textDistance <= maxViewDistance
```

Comment:
Just compare with cosine of TEXT_ELEMENT_FALLOFF_ANGLE so you would not need to compute the acos every frame

Reply:
Also the return value should be undefined if the label is too far away. You are returning a bool